packages/dynamite/lib/src/openapi_builder.dart

```diff
@@ -1268,7 +1268,9 @@ TypeResult resolveObject(
                     final isContentStringArray = isContentString && result is TypeResultList;
                     final hasExtraJsonKeyValues =
                         extraJsonKeyValues != null && extraJsonKeyValues.containsKey(propertyName);
-                    if (hasDifferentName || isContentString || isContentStringArray || hasExtraJsonKeyValues) {
+                    if (hasDifferentName ||
+                        result is! TypeResultBase && (isContentString || isContentStringArray) ||
```

Comment:
Only checking for `TypeResultBase` is not enough, since `TypeResultEnum` is also just a string and `TypeResultList` is also valid in this case. I think you should invert the condition and only check against `TypeResultMap` and `TypeResultObject`.

Reply:
Although `TypeResultMap` would have the same problems with `.fromJsonString` and `.toJsonString`, right? But it should definitely work with that... No idea how to fix that